[T13-5] Workoutbook

yicheng12 opened the pull request.

devamanyu commented on docs/DeveloperGuide.adoc:
```diff
@@ -1,4 +1,4 @@
-= AddressBook Level 4 - Developer Guide
+= Workout Book - Developer Guide
```
Consider changing the team name and dates to reflect your team

devamanyu commented on docs/DeveloperGuide.adoc:
```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+==== Aspect: How recommend executes
+
+* **Alternative 1 (current choice):** Filters the workout list based on user's input and randomly generates a workout from that filtered list.
```
Is the recommended workout "retrieved" randomly or "generated" randomly? Specify this to remove confusion.

devamanyu commented on docs/DeveloperGuide.adoc:
```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+[[fig-ProfileCommand]]
+.Interactions inside `Logic` component for the *`profile`* command
+image::ProfileCommand.PNG[width="800"]
```
I am confused about some of the events in this sequence diagram.  1) If showProfileRequestEvent() is shown to create an "Events" object, then shouldn't there be a constructor. 2) Some of the processes seem to be working in "parallel", e.g. CommandResult is invoked before control returns from "Events". This may not be true as the code that you are working with are working sequentially.  Do give a look into it. Please ignore, if I understood wrongly.

devamanyu commented on docs/DeveloperGuide.adoc:
```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+// tag::trackfeature[]
+=== Tracking Feature
+==== Current Implementation
```
This section seems to be incomplete. Please consider completing it.

devamanyu commented on docs/AboutUs.adoc:
```diff
@@ -19,38 +17,47 @@ Role: Project Advisor
 
```
Please remove Prof. Damith's photo from this page.

devamanyu reviewed (commented):
Overall a nice effort on the documentation. Its heartening to see that UML diagrams were attempted for each feature. 

I have provided some comments, do give a look.

yicheng12 committed:
update ProfileWindow.html

yicheng12 committed:
update ProfileWindow.html

yicheng12 committed:
update ProfileWindow.html

yicheng12 committed:
update ProfileWindow.html

sijieg committed:
Fix developer guide

sijieg committed:
fix travis errors

yicheng12 committed:
update ProfileWindow.html

sijieg committed:
Merge pull request #231 from sijieg/master

fix travis error

yicheng12 committed:
merge conflicts

yicheng12 committed:
fix travis errors

yicheng12 committed:
fix travis errors

yicheng12 committed:
fix travis errors

shouteck committed:
Update PPP and Refactor existing code.

shouteck committed:
Fix travis errors

yicheng12 committed:
update PPP

shouteck committed:
Update PPP

shouteck committed:
Update PPP

shouteck committed:
Merge pull request #233 from yicheng12/master

update PPP

shouteck committed:
Merge remote-tracking branch 'origin/master'

sijieg committed:
Fix some test case

sijieg committed:
Fix travis errors

sijieg committed:
Merge pull request #235 from shouteck/master

Update PPP and refactor some codes

sijieg committed:
Merge remote-tracking branch 'upstream/master'

sijieg committed:
Fix travis errors

shouteck committed:
Update DeveloperGuide

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

shouteck committed:
Merge pull request #236 from sijieg/master

Fix some test case

DarylTew committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	ProfileWindow.html
#	src/main/java/seedu/address/model/workout/Workout.java

sijieg committed:
Fix a minor bug

sijieg committed:
Fix a minor bug

sijieg committed:
Update UserGuide and generated more workouts at the beginning.

sijieg committed:
Merge pull request #237 from sijieg/master

Fix a minor bug

shouteck committed:
Update DeveloperGuide

shouteck committed:
Update DeveloperGuide

shouteck committed:
Merge remote-tracking branch 'origin/master'

shouteck committed:
Update DeveloperGuide

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide

sijieg committed:
Update UserGuide and Developer Guide

Change keypress for profile from F2 to F3
Added event for current command to select the workout that is tag current if successful
Changed userprofilecard

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide

shouteck committed:
Fix travis

sijieg committed:
Merge pull request #239 from shouteck/master

Update UserGuide

sijieg committed:
Merge remote-tracking branch 'upstream/master'

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

sijieg committed:
Update developer guide

sijieg committed:
Fix travis error

sijieg committed:
Merge pull request #240 from sijieg/master

 Update UserGuide and Developer Guide

sijieg committed:
Fix a small typo.

sijieg committed:
Merge pull request #241 from sijieg/master

Fix a small typo.

sijieg committed:
Fix a small typo.

sijieg committed:
Merge pull request #242 from sijieg/master

Fix a small typo.

DarylTew committed:
Merge remote-tracking branch 'origin/master'

DarylTew committed:
Fix headless error

DarylTew committed:
Edit CurrentCommand and CurrentCommandTest

CurrentCommand now checks every workout for presence of current state
tag

Minor fix to typo in FilterCommand

sijieg committed:
Fix some typo in Developer Guide.

sijieg committed:
Fix some typo in Developer Guide.

sijieg committed:
Merge pull request #243 from sijieg/master

Fix some typo in Developer Guide

DarylTew committed:
Merge remote-tracking branch 'origin/master'

DarylTew committed:
Fix EditCommand

where tag/ does not give an error when there is only the state tag left

shouteck committed:
Fix travis

sijieg committed:
Merge pull request #244 from DarylTew/master

 Edit CurrentCommand and CurrentCommandTest and Fix EditCommand

sijieg committed:
Fix AddCommandSystemTest

sijieg committed:
Add more test cases for SystemTests

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

garyheng committed:
Fix TrackCommandParser's parse method

Fix:
1. parse() now checks that the VALUE entered for `track` is one string/value with no spaces

garyheng committed:
Merge pull request #245 from sijieg/master

 Fix AddCommandSystemTest

garyheng committed:
Add WorkoutContainsParameterPredicateTest

garyheng committed:
Fix TrackCommandParserTest

Fix:
1. TrackCommandParserTest tests now contains correct test cases that match the value constraints set in `track`

garyheng committed:
Add VersionedTrackedDataTest

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

garyheng committed:
Add TrackedDataTest

garyheng committed:
Fix several tests

Fix:
1. LogicManagerTest to cover all methods in LogicManager
2. typographical errors in SelectCommandTest

garyheng committed:
Add XmlAdaptedParameterTest

garyheng committed:
Fix XmlAdaptedWorkout, XmlSerializableWorkoutBookTest

Fix:
1. XmlAdaptedWorkout to match instruction validity
2. XmlSerializableWorkoutBookTest to include previously removed test

garyheng committed:
Add XmlSerializableTrackedDataListTest

garyheng committed:
Add XmlSerializableTrackedDataTest

garyheng committed:
Add test methods to StorageManagerTest

garyheng committed:
Add XmlTrackedDataStorageTest

garyheng committed:
Add XmlTrackedDataListStorageTest, Fix multiple checkstyle errors

Fix checkstyle errors in:
1. TrackCommandParserTest
2. TrackedDataTest
3. XmlSerializableTrackedDataListTest
4. XmlTrackedDataStorageTest
5. TypicalParameters

garyheng committed:
Fix checkstyle error in XmlTrackedDataListStorageTest

DarylTew committed:
Add CompleteCommandTest

garyheng committed:
Fix SelectCommandSystemTest and typographical errors in several files

garyheng committed:
Fix TrackCommandTest

sijieg committed:
Update README.adoc

sijieg committed:
Merge pull request #246 from sijieg/master

Update README.adoc

garyheng committed:
Fix errors in SelectCommandSystemTest

sijieg committed:
Update README.adoc

sijieg committed:
Merge pull request #247 from sijieg/master

Update ppp

DarylTew committed:
Update daryltew.adoc

DarylTew committed:
Merge remote-tracking branch 'origin/master'

DarylTew committed:
Merge pull request #248 from DarylTew/master

 Add CompleteCommandTest

DarylTew committed:
Delete CompleteCommandSD.png

DarylTew committed:
Delete CurrentCommandSD.png

DarylTew committed:
Add CurrentCommand and CompleteCommand SD

DarylTew committed:
Update DeveloperGuide.adoc

DarylTew committed:
Update README.adoc

DarylTew committed:
Update README.adoc

DarylTew committed:
Update DeveloperGuide.adoc

garyheng committed:
Add test cases to several tests

Add test cases to:
1. TrackCommandParserTest
2. WorkoutContainsParameterPredicateTest
3. XmlAdaptedParameterTest
4. TrackedDataListPanelTest

Add:
1. CurrentAndTrackPanelTest

garyheng committed:
Fix checkstyle error in XmlAdaptedParameterTest

yicheng12 committed:
update DeveloperGuide and yicheng12.doc

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

yicheng12 committed:
fix travis error with EOF

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

garyheng committed:
Merge branch 'master' into track-v1.4

garyheng committed:
Update UserGuide, DeveloperGuide and Project Portfolio

garyheng committed:
Merge pull request #249 from yicheng12/master

update developerGuide and yicheng12.adoc

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

garyheng committed:
Merge branch 'track-v1.4'

garyheng committed:
Fix DeveloperGuide

DarylTew committed:
Update daryltew.adoc

yicheng12 committed:
Merge pull request #250 from garyheng/master

Add and fix multiple tests, Update guides

yicheng12 committed:
update ppp

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

yicheng12 committed:
update ppp

yicheng12 committed:
update ppp and DeveloperGuide.adoc

garyheng committed:
Fix PPP, UserGuide and DeveloperGuide

garyheng committed:
Fix UserGuide

garyheng committed:
Fix Project PortFolio

garyheng committed:
Merge pull request #252 from yicheng12/master

update ppp and developerGuide

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

garyheng committed:
Fix Project Portfolio

garyheng committed:
Merge pull request #253 from garyheng/master

Update Project Portfolio and UserGuide

garyheng committed:
Update UserGuide.adoc

garyheng committed:
Merge pull request #254 from garyheng/master

Update UserGuide

DarylTew committed:
Update AboutUs.adoc

DarylTew committed:
Update AboutUs.adoc

DarylTew committed:
Update UserGuide.adoc

DarylTew committed:
Update daryltew.adoc

DarylTew committed:
Update DeveloperGuide.adoc

yicheng12 committed:
update AboutUs.adoc

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

sijieg committed:
Update ModifyCommandTest and ProfileWindowManagerTest

sijieg committed:
Merge remote-tracking branch 'upstream/master'

sijieg committed:
Merge pull request #255 from yicheng12/master

update AboutUs.adoc

sijieg committed:
Merge remote-tracking branch 'upstream/master'

sijieg committed:
Fix travis errors

sijieg committed:
Update developer guide

sijieg committed:
Improve test cases

sijieg committed:
Fix travis errors

sijieg committed:
Improve test cases

sijieg committed:
Fix travis errors

sijieg committed:
Improve test cases

sijieg committed:
Fix travis errors

yicheng12 committed:
create sortCommandtest

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

sijieg committed:
Improve test cases

sijieg committed:
Fix travis errors

shouteck committed:
Merge remote-tracking branch 'origin/master'

DarylTew committed:
Merge pull request #257 from yicheng12/master

create sortCommandTest

DarylTew committed:
Update DeveloperGuide.adoc

sijieg committed:
Merge remote-tracking branch 'upstream/master'

DarylTew committed:
Update DeveloperGuide.adoc

DarylTew committed:
Update daryltew.adoc

sijieg committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

DarylTew committed:
Update DeveloperGuide.adoc

sijieg committed:
Improve test cases

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

shouteck committed:
Merge remote-tracking branch 'origin/master'

shouteck committed:
Update RecommendCommand and RecommendCommandIntegration tests.
Update PPP and DeveloperGuide

shouteck committed:
Minor fix in RecommendCommandIntegrationTest

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Fix travis errors

sijieg committed:
Merge remote-tracking branch 'upstream/master'

DarylTew committed:
Merge pull request #258 from sijieg/master

Update Test cases

sijieg committed:
test

sijieg committed:
Create config.json

sijieg committed:
Merge pull request #259 from sijieg/master

change config.json

sijieg committed:
update PPP

sijieg committed:
Merge remote-tracking branch 'upstream/master'

sijieg committed:
update PPP

sijieg committed:
Merge pull request #260 from sijieg/master

Update PPP

DarylTew committed:
Add descriptions to FilterCommandParser and RemarkCommandParser

DarylTew committed:
Update daryltew.adoc

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

shouteck committed:
Merge pull request #261 from DarylTew/master

Update PPP

shouteck committed:
Minor fix in RecommendCommandIntegrationTest

shouteck committed:
Merge remote-tracking branch 'origin/master'

shouteck committed:
Minor fix in RecommendCommandIntegrationTest

sijieg committed:
Merge pull request #262 from shouteck/master

Minor fix in RecommendCommandIntegrationTest

sijieg committed:
update PPP

sijieg committed:
Merge pull request #263 from sijieg/master

update ReadME

sijieg committed:
update PPP

sijieg committed:
Merge pull request #264 from sijieg/master

Update Readme

sijieg committed:
update PPP

sijieg committed:
Merge pull request #265 from sijieg/master

update PPP

sijieg committed:
Update CurrentCommand

shouteck committed:
Update DeveloperGuide

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

yicheng12 committed:
Merge pull request #266 from sijieg/master

Update CurrentCommand

sijieg committed:
Update UserGuide

yicheng12 committed:
add 10 more sample workouts

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

yicheng12 committed:
update yicheng12.adoc

DarylTew committed:
Merge pull request #267 from sijieg/master

Update UserGuide

yicheng12 committed:
fix travis errors

yicheng12 committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

yicheng12 committed:
fix travis errors

yicheng12 committed:
fix travis errors

sijieg committed:
Merge pull request #268 from yicheng12/master

update SampleDataUtil

sijieg committed:
Update DeveloperGuide

sijieg committed:
Merge pull request #269 from sijieg/master

Update DeveloperGuide

shouteck committed:
Merge remote-tracking branch 'origin/master'

sijieg committed:
Add files via upload

sijieg committed:
Merge pull request #270 from CS2113-AY1819S1-T13-5/sijieg-patch-1

Add ModifyCommandSD.PNG

sijieg committed:
Update DeveloperGuide

sijieg committed:
Update DeveloperGuide

sijieg committed:
Update DeveloperGuide

sijieg committed:
Merge pull request #271 from sijieg/master

 Update DeveloperGuide

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide

shouteck committed:
Update UserGuide and DeveloperGuide

shouteck committed:
Update DeveloperGuide

shouteck committed:
Update UserGuide

shouteck committed:
Fix travis

sijieg committed:
Merge pull request #273 from shouteck/master

Update Developer and User Guide

shouteck committed:
Update Developer Guide

shouteck committed:
Merge pull request #274 from shouteck/master

Update Developer Guide

garyheng committed:
Fix UserGuide and DeveloperGuide

garyheng committed:
Merge branch 'master' of https://github.com/CS2113-AY1819S1-T13-5/main

sijieg committed:
Merge pull request #275 from garyheng/master

Fix UserGuide and DeveloperGuide

garyheng committed:
FIx UserGuide

sijieg committed:
Merge pull request #276 from garyheng/master

FIx UserGuide

DarylTew committed:
Update daryltew.adoc

DarylTew committed:
Fix Typo

garyheng committed:
Fix Project Portfolio

garyheng committed:
Merge pull request #277 from garyheng/master

Fix Project Portfolio

DarylTew committed:
Merge remote-tracking branch 'origin/master'

DarylTew committed:
Fix Typo

DarylTew committed:
Update DeveloperGuide.adoc

DarylTew committed:
Update UserGuide.adoc

DarylTew committed:
Merge pull request #278 from DarylTew/master

Fix guides

shouteck committed:
Update AboutUs.adoc

DarylTew committed:
Update DeveloperGuide.adoc

DarylTew committed:
Update DeveloperGuide.adoc

shouteck committed:
Delete docs directory

Deleted for privacy concerns
